07_purity_prediction/feature_robustness.ipynb

```diff
@@ -0,0 +1,516 @@
+{
```

Comment:
<p>Are these results what you expected? I guess I'm not sure what I'd expect the distribution to look like ahead of time, but isn't it interesting that genes are either always useful or never useful regardless of cancer type? </p><p> </p><p>Doesn't this imply that in the pancancer f-test feature selection ~40% of your features could be thrown out and still see the same performance?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/pancancer-evaluation/pull/50/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<blockquote>Are these results what you expected?</blockquote><p> </p><p>Yeah, these are more or less what I expected. These results are for the "all other cancers" cross-validation splits, which means that if we're testing on cancer type X, we're training on all other cancer types in TCGA minus X. Since there are 33 cancer types in TCGA, this means the training set isn't changing too much between cancer types and cross-validation folds, so I'd expect the features selected to be pretty consistent for all the non-random selection methods.</p><p> </p><p>At some point I should probably do this for the single cancer cross-validation splits, where I would expect the features selected to vary quite a bit more because the training sets are completely different for each cancer type.</p><p> </p><blockquote>Doesn't this imply that in the pancancer f-test feature selection ~40% of your features could be thrown out and still see the same performance?</blockquote><p> </p><p>I guess so, but it's hard to know a priori which 40%. We already know from past experiments (and other studies have shown the same thing) that many small subsets of features can achieve similar performance on the same dataset for gene expression, since there's so much redundancy/correlation in the data.</p><p> </p><p>Our goal with this study is to determine if there's a simple way to select "general" features -- i.e. from the redundant ones, select ones that are more likely to generalize outside of the training dataset, or that capture general cancer processes, etc. This plot just shows that those "general" features aren't necessarily more or less consistent than selecting features just based on variance (MAD), which isn't terribly surprising but is interesting to know.</p>